WrightTools/kit.py

```diff
@@ -1056,6 +1056,40 @@ def __init__(self, xi, yi, k=3, s=1000, ignore_nans=True):
         self.true_spline = UnivariateSpline(xi_internal, yi_internal, k=k, s=s)
 
 
+def symmetric_sqrt(x, out=None):
+    """Compute the 'symmetric' square root: sign(x) * sqrt(abs(x)).
+
+    Parameters
+    ----------
+    x : array_like or number
+        Input array.
+    out : ndarray, None, or tuple of ndarray and None (optional)
+        A location into which the result is stored. If provided, it must
+        have a shape that the inputs broadcast to. If not provided or None,
+        a freshly-allocated array is returned. A tuple (possible only as a
+        keyword argument) must have length equal to the number of outputs.
+
+    Returns
+    -------
+    np.ndarray
+        Symmetric square root of arr.
+    """
+    if hasattr(x, '__len__'):
+        x = np.array(x)
+        number = False
+    else:
+        x = np.array([x])
+        number = True
+    factor = np.ones(x.shape)
+    factor[x < 0] = -1
```

Comment:
please use [np.sign](https://docs.scipy.org/doc/numpy/reference/generated/numpy.sign.html)

This will also allow you to transparently pass numbers, without any if/else blocks

Reply:
I just realized your test is exactly that... Why all the overhead in the function?